nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Discover/SdkProjectDiscovery.cs

```diff
@@ -109,6 +109,7 @@ public static async Task<ImmutableArray<ProjectDiscoveryResult>> DiscoverWithBin
                         $"/p:TargetFramework={tfm}",
                         $"/p:CustomAfterMicrosoftCommonCrossTargetingTargets={dependencyDiscoveryTargetsPath}",
                         $"/p:CustomAfterMicrosoftCommonTargets={dependencyDiscoveryTargetsPath}",
+                        "/p:TreatWarningsAsErrors=false", // if using CPM and a project also sets TreatWarningsAsErrors to true, this can cause discovery to fail; explicitly don't allow that
```

Comment:
This was the fix.

Reply:
We also have `MSBuildTreatWarningsAsErrors` set to true, I don't know enough about the different here to know if its relevant 

https://learn.microsoft.com/en-us/visualstudio/msbuild/common-msbuild-project-properties?view=vs-2022

Reply:
Fixed, and thank you for checking.

Reply:
Would something like `/p:NoWarn=$(NoWarn);NU1507` be a more targeted fix? I'm concerned that bulk turning off WarningsAsErrors could allow us to slip verifiably broken builds into PRs.

Reply:
Talked offline, sounds like NoWarn can't be passed as a command line arg and there are additional correctness considerations.